OpenRA.Game/Input/Keycode.cs

```diff
@@ -259,254 +259,255 @@ public enum Keycode
+			{ Keycode.KP_DBLVERTICALBAR, "keycode.kp_dblverticalbar" },
+			{ Keycode.KP_COLON, "keycode.kp_colon" },
+			{ Keycode.KP_HASH, "keycode.kp_hash" },
+			{ Keycode.KP_SPACE, "keycode.kp_space" },
+			{ Keycode.KP_AT, "keycode.kp_at" },
+			{ Keycode.KP_EXCLAM, "keycode.kp_exclam" },
+			{ Keycode.KP_MEMSTORE, "keycode.kp_memstore" },
+			{ Keycode.KP_MEMRECALL, "keycode.kp_memrecall" },
+			{ Keycode.KP_MEMCLEAR, "keycode.kp_memclear" },
+			{ Keycode.KP_MEMADD, "keycode.kp_memadd" },
+			{ Keycode.KP_MEMSUBTRACT, "keycode.kp_memsubtract" },
+			{ Keycode.KP_MEMMULTIPLY, "keycode.kp_memmultiply" },
+			{ Keycode.KP_MEMDIVIDE, "keycode.kp_memdivide" },
+			{ Keycode.KP_PLUSMINUS, "keycode.kp_plusminus" },
+			{ Keycode.KP_CLEAR, "keycode.kp_clear" },
+			{ Keycode.KP_CLEARENTRY, "keycode.kp_clearentry" },
+			{ Keycode.KP_BINARY, "keycode.kp_binary" },
+			{ Keycode.KP_OCTAL, "keycode.kp_octal" },
+			{ Keycode.KP_DECIMAL, "keycode.kp_decimal" },
+			{ Keycode.KP_HEXADECIMAL, "keycode.kp_hexadecimal" },
+			{ Keycode.LCTRL, "keycode.lctrl" },
+			{ Keycode.LSHIFT, "keycode.lshift" },
+			{ Keycode.LALT, "keycode.lalt" },
+			{ Keycode.LGUI, "keycode.lgui" },
+			{ Keycode.RCTRL, "keycode.rctrl" },
+			{ Keycode.RSHIFT, "keycode.rshift" },
+			{ Keycode.RALT, "keycode.ralt" },
+			{ Keycode.RGUI, "keycode.rgui" },
+			{ Keycode.MODE, "keycode.mode" },
+			{ Keycode.AUDIONEXT, "keycode.audionext" },
+			{ Keycode.AUDIOPREV, "keycode.audioprev" },
+			{ Keycode.AUDIOSTOP, "keycode.audiostop" },
+			{ Keycode.AUDIOPLAY, "keycode.audioplay" },
+			{ Keycode.AUDIOMUTE, "keycode.audiomute" },
+			{ Keycode.MEDIASELECT, "keycode.mediaselect" },
+			{ Keycode.WWW, "keycode.www" },
+			{ Keycode.MAIL, "keycode.mail" },
+			{ Keycode.CALCULATOR, "keycode.calculator" },
+			{ Keycode.COMPUTER, "keycode.computer" },
+			{ Keycode.AC_SEARCH, "keycode.ac_search" },
+			{ Keycode.AC_HOME, "keycode.ac_home" },
+			{ Keycode.AC_BACK, "keycode.ac_back" },
+			{ Keycode.AC_FORWARD, "keycode.ac_forward" },
+			{ Keycode.AC_STOP, "keycode.ac_stop" },
+			{ Keycode.AC_REFRESH, "keycode.ac_refresh" },
+			{ Keycode.AC_BOOKMARKS, "keycode.ac_bookmarks" },
+			{ Keycode.BRIGHTNESSDOWN, "keycode.brightnessdown" },
+			{ Keycode.BRIGHTNESSUP, "keycode.brightnessup" },
+			{ Keycode.DISPLAYSWITCH, "keycode.displayswitch" },
+			{ Keycode.KBDILLUMTOGGLE, "keycode.kbdillumtoggle" },
+			{ Keycode.KBDILLUMDOWN, "keycode.kbdillumdown" },
+			{ Keycode.KBDILLUMUP, "keycode.kbdillumup" },
+			{ Keycode.EJECT, "keycode.eject" },
+			{ Keycode.SLEEP, "keycode.sleep" },
+			{ Keycode.MOUSE4, "keycode.mouse4" },
+			{ Keycode.MOUSE5, "keycode.mouse5" },
 		};
 
 		public static string DisplayString(Keycode k)
 		{
-			if (!KeyNames.TryGetValue(k, out var ret))
+			if (!KeycodeTranslationKeys.TryGetValue(k, out var translationKey))
 				return k.ToString();
 
-			return ret;
+			return TranslationProvider.GetString(translationKey);
```

Comment:
This should also be cached, this is only used in Hotkey so might be better to move it there for LoB